storetests/local/localstore_test.go

```diff
@@ -25,7 +26,50 @@ func TestLocalStoreCompressedZst(t *testing.T) {
 	storetests.TestAll(t, createlocalStoreFactory(t, "zstd"))
 }
 
-func createlocalStoreFactory(t *testing.T, compression string) storetests.StoreFactory {
+func TestLocalStore_CompressionAndMetering(t *testing.T) {
+	compressedReadByteCount := 0
+	compressedWriteByteCount := 0
+	uncompressedReadByteCount := 0
+	uncompressedWriteByteCount := 0
+
+	var uncompressedRead string
+	var compressedRead string
+	var compressedWrite string
+	var uncompressedWrite string
+
+	opts := []dstore.Option{
+		dstore.WithCompressedReadCallback(func(ctx context.Context, i int) {
+			compressedReadByteCount += i
+			compressedRead = "compressedRead"
+		}),
+		dstore.WithUncompressedReadCallback(func(ctx context.Context, i int) {
+			uncompressedReadByteCount += i
+			uncompressedRead = "uncompressedRead"
+		}),
+		dstore.WithCompressedWriteCallback(func(ctx context.Context, i int) {
+			compressedWriteByteCount += i
+			compressedWrite = "compressedWrite"
+		}),
+		dstore.WithUncompressedWriteCallback(func(ctx context.Context, i int) {
+			uncompressedWriteByteCount += i
+			uncompressedWrite = "uncompressedWrite"
+		}),
+	}
+
+	storetests.TestAll(t, createlocalStoreFactory(t, "zstd", opts...))
```

Comment:
It's a bit abusive to re-run all the tests, they take relatively long time to run. But I can understand it was easier like this.

I don't want to block the merge of this no more, so let's go with this, it's not like we run this integration tests that often.